lib/mutant/matcher/compiler.rb

```diff
@@ -34,60 +34,23 @@ def call(subject)
 
     private
 
-      # Predicate constraining matches
+      # Predicate returning false on ignored subject
       #
       # @return [#call]
       #
-      # rubocop:disable MethodLength
-      #
       # @api private
-      def predicate
-        if subject_selector && subject_rejector
-          Morpher::Evaluator::Predicate::Boolean::And.new([
-            subject_selector,
-            Morpher::Evaluator::Predicate::Negation.new(subject_rejector)
-          ])
-        elsif subject_selector
-          subject_selector
-        elsif subject_rejector
-          Morpher::Evaluator::Predicate::Negation.new(subject_rejector)
+      def ignored_subjects
+        rejectors = config.ignore_expressions.map(&SubjectPrefix.method(:new))
+
+        if rejectors.any?
+          Morpher::Evaluator::Predicate::Boolean::Negation.new(
+            Morpher::Evaluator::Predicate::Boolean::Or.new(rejectors)
+          )
```

Comment:
I'm not sure how your predicate objects work, but one general pattern that works here without a conditional would be:

```ruby
predicate = rejectors.inject(Contradiction.new) do |predicate, rejector|
  Or.new(predicate, rejector)
end
Negation.new(predicate)
```

Obviously the resulting object graph is a bit larger, but it removes the need for conditional logic.